### PR TITLE
Imt/felipe

### PR DESCRIPTION
**Pull request by felp99**
For following changes

### CODE REVIEW

File: app/[locale]/(dashboard)/sensors/[id]/upload/page.jsx [hunk 1]
```diff
@@ -11,9 +11,12 @@ export default async function Home({ params }) {
       id: true,
       identifier: true,
       station: true,
+      files: true,
     },
   })
 
+  console.log(sensor)
```
Comment (felp99) [inline app/[locale]/(dashboard)/sensors/[id]/upload/page.jsx]: take this out

File: app/api/sensors/route.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { NextResponse } from 'next/server'
+import { S3 } from '@aws-sdk/client-s3'
+import { Upload } from '@aws-sdk/lib-storage'
+
+const uploadFile = async function (file, file_id, sensor_id) {
+  const config = {
+    region: process.env.REGION || 'fr-par',
+    endpoint: process.env.ENDPOINT || 'https://s3.fr-par.scw.cloud',
+    credentials: {
+      accessKeyId: process.env.SCW_ACCESS_KEY_ID,
```
Comment (cballevre) [inline app/api/sensors/route.js]: I think we should add `SCW_ACCESS_KEY_ID` and `SCW_SECRET_ACCESS_KEY` in the env.example so people can understand they need to set them

File: app/api/sensors/route.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      name: _file.name,
+      status: 'Active',
+      created_at: new Date(),
+      sensor_id: sensor_id,
+    },
+  })
+  return file.id
+}
+
+export async function POST(request) {
```
Comment (cballevre) [inline app/api/sensors/route.js]: The POST route is already use to create new sensor, I think you should use an other route to upload the file

File: app/api/sensors/route.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import { db } from '@/lib/db'
+import { NextResponse } from 'next/server'
+import { S3 } from '@aws-sdk/client-s3'
+import { Upload } from '@aws-sdk/lib-storage'
+
+const uploadFile = async function (file, file_id, sensor_id) {
```
Comment (cballevre) [inline app/api/sensors/route.js]: Maybe you can keep this code into the lib aws

File: components/Form/SensorsUploadForm.jsx [hunk 5]
```diff
@@ -17,11 +17,23 @@ export default function SensorsUploadForm({ sensor }) {
...
     if (event.target.files && event.target.files[0]) {
       const i = event.target.files[0]
       setFormData({ ...formData, file: i })
-      uploadFile(i)
+
+      var fd = new FormData()
+      fd.append('file', i)
+      fd.append('sensor_id', sensor.id)
+
+      try {
```
Comment (cballevre) [inline components/Form/SensorsUploadForm.jsx]: You can use react-query to manage api call so its easier to manage error

File: components/Form/SensorsUploadForm.jsx [hunk 6]
```diff
@@ -45,10 +57,12 @@ export default function SensorsUploadForm({ sensor }) {
         label={t('upload_screen.upload_file_for_sensor')}
       />
       {formData.file !== null && (
-        <div>
-          <Typography>{formData.file.name}</Typography>
-          <Typography>{formData.file.size}</Typography>
-        </div>
+        <>
```
Comment (cballevre) [inline components/Form/SensorsUploadForm.jsx]: You don't need a need fragment here I think the div is good enough

**Review by cballevre — APPROVED** [review]
Nice feature before improving it I would merge with the rest of the project

File: app/api/sensors/route.js (hunk 3 above)
Comment (felp99) [reply app/api/sensors/route.js]: Ok, updated this

**Pull request merged**